src/examples/hotgym/HelloSPTP.cpp

```diff
@@ -115,8 +123,11 @@ Real64 BenchmarkHotgym::run(UInt EPOCHS, bool useSPlocal, bool useSPglobal, bool
 
     //Encode
     tEnc.start();
-    enc.encodeIntoArray(r, input.data());
+    enc.encode(r, inputSDR);
     tEnc.stop();
+    for(auto i = 0u; i < inputSDR.size; ++i) {
+      input[i] = (UInt) inputSDR.getDense()[i];
```

Comment:
nit, you could use VectorHelpers::castVectorType<Byte, UInt>(inputSDR.data()) as used on other places of this PR. 
PS: I'll be trying to avoid that by templating the SDR_dense_t 's datatype in SDR in another PR.

Reply:
I think it would be far easier & better to overhaul this example to use SDRs throughout.

Reply:
Once encoders support SDR, it should go there. So it could be part of this PR.